reconcile/test/test_utils_terrascript_aws_client.py

```diff
@@ -49,27 +50,79 @@ def is_job_running(self, name):
 
 
 def test_use_previous_image_id_no_upstream(ts):
-    assert ts._use_previous_image_id({}) is False
+    assert ts._use_previous_image_id([]) is False
 
 
-def test_use_previous_image_id_false(mocker, ts):
-    result = False
+@pytest.mark.parametrize("result", [True, False])
+def test_use_previous_image_id(mocker, ts, result):
     mocker.patch(
         "reconcile.utils.terrascript_aws_client.TerrascriptClient.init_jenkins",
         return_value=MockJenkinsApi(result),
     )
-    image = {"upstream": {"instance": {"name": "ci"}, "name": "job"}}
+    image = [{"upstream": {"instance": {"name": "ci"}, "name": "job"}}]
     assert ts._use_previous_image_id(image) == result
 
 
-def test_use_previous_image_id_true(mocker, ts):
-    result = True
+def test_get_asg_image_id(mocker, ts: tsclient.TerrascriptClient):
```

Comment:
hmm, if you're writing tests for it, should `_get_asg_image_id` be public?

Reply:
Why should it be public then? Currently, `_get_asg_image_id` is just used internally, making sense to be private.

Reply:
I understand that this is an internal method but I'm usually under the assumption that private methods are implementation details and there's no expectation of how they are going to behave exactly and they can be refactored without any fear. The moment you create a test, you create an expectation of it staying in a certain way. Hence my point.

Maybe others will disagree so I'm happy to accept this as it is if people don't mind it. I won't die on this hill 😂

Reply:
I'll make it public, and you can sleep in peace :)